common/types/client.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"context"
+	"math/big"
+)
+
+type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
```

Comment:
Instead of using any for all types here, could you use the specific constraint type for things that we have already defined?
For example:
ADDR types.Hashable[ADDR]
TX_HASH types.Hashable[TX_HASH]
BLOCK_HASH types.Hashable[BLOCK_HASH]


Also, if possible, could you look at my latest PR on the new types introduced there?
(https://github.com/smartcontractkit/chainlink/pull/8982)
Do you think we could merge that PR before yours, and use those types in your PR? If you think that's too much work for you, then its fine to ignore that PR. We will then merge your PR first, and then incorporate that one later.

Reply:
That's totally fine. Let me clean up the mess and I'll rebase on your PR instead.